x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/accesscontrol/DocumentSubsetReader.java

```diff
@@ -86,7 +86,7 @@ private static int computeNumDocs(LeafReader reader, BitSet roleQueryBits) {
     private static int getNumDocs(LeafReader reader, Query roleQuery, BitSet roleQueryBits) throws IOException, ExecutionException {
         IndexReader.CacheHelper cacheHelper = reader.getReaderCacheHelper(); // this one takes deletes into account
         if (cacheHelper == null) {
-            throw new IllegalStateException("Reader " + reader + " does not support caching");
+            return computeNumDocs(reader, roleQueryBits);
```

Comment:
This change was necessary as DocumentSubsetReader was a bad citizen. It required the leafreader that it wrapped to have a readerCacheHelper, but then at the same time did not expose a reader cache helper by its own wrapping leafreader, so essentially double-standards.

I don't see a reason for this reader not to work when caching isn't available (single-document case)